File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3098_find-the-sum-of-subsequence-powers/solutions/cpp/solution.cpp
#include <vector>
#include <algorithm>
#include <numeric>

using namespace std;

class Solution {
public:
    int sumOfPowers(vector<int>& nums, int k) {
        const int MOD = 1000000007;
        sort(nums.begin(), nums.end());
        int n = nums.size();

        // Prepare combinations calculations
        vector<vector<long long>> comb(n + 1, vector<long long>(k + 1, 0));
        for (int i = 0; i <= n; i++) {
            comb[i][0] = 1;
            for (int j = 1; j <= min(i, k); j++) {
                comb[i][j] = (comb[i-1][j] + comb[i-1][j-1]) % MOD;
            }
        }

        long long totalPower = 0;

        // Calculate the contribution of each difference
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                long long diff = nums[j] - nums[i];
                // Left choices * right choices
                long long leftChoices = comb[i][k-2];
                long long rightChoices = comb[n-j-1][k-2];
                totalPower = (totalPower + diff % MOD * leftChoices % MOD * rightChoices % MOD) % MOD;
            }
        }

        return (int) totalPower;
    }
};

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3098_find-the-sum-of-subsequence-powers/solutions/python3/solution.py
from itertools import combinations

class Solution:
    def sumOfPowers(self, nums: list[int], k: int) -> int:
        MOD = 10**9 + 7
        nums.sort()
        n = len(nums)
        result = 0
        
        for combo in combinations(nums, k):
            # Minimum absolute difference in this subsequence
            min_diff = min(abs(combo[i] - combo[i-1]) for i in range(1, k))
            result = (result + min_diff) % MOD
            
        return result
